bugshaker/src/main/java/com/github/stkent/bugshaker/MainActivity.java

```diff
@@ -0,0 +1,217 @@
+package com.github.stkent.bugshaker;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.util.Set;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.MediaStore;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.Toast;
+
+import com.github.stkent.bugshaker.flow.widget.DrawingView;
+import com.github.stkent.bugshaker.utilities.LogcatUtil;
+import com.github.stkent.bugshaker.utilities.SendEmailUtil;
+import com.github.stkent.bugshaker.utilities.SharedPreferencesUtil;
+
+public class MainActivity extends AppCompatActivity implements OnClickListener {
+
+	private DrawingView drawView;
+
+	private static final String tag = "MainActivity";
+
+	private ImageButton currPaint;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_main);
+
+		drawView = (DrawingView) findViewById(R.id.drawing);
+		LinearLayout paintLayout = (LinearLayout) findViewById(R.id.paint_colors);
+		currPaint = (ImageButton) paintLayout.getChildAt(0);
+		currPaint.setImageDrawable(getResources().getDrawable(R.drawable.paint_pressed));
+
+		ImageButton drawBtn = (ImageButton) findViewById(R.id.draw_btn);
+		drawBtn.setOnClickListener(this);
+
+		drawView.setBrushSize(getResources().getInteger(R.integer.small_size));
+
+		findViewById(R.id.erase_btn).setOnClickListener(this);
+		findViewById(R.id.sendEmail).setOnClickListener(this);
+		findViewById(R.id.speechBox).setOnClickListener(this);
+
+		String pathOfScreenshot = getIntent().getStringExtra("uri");
+		File screenshotFile = new File(pathOfScreenshot);
+
+
+		if (screenshotFile.exists()) {
+			Bitmap myBitmap = BitmapFactory.decodeFile(pathOfScreenshot);
+
```

Comment:
This logic shouldn't all be inside onCreate. We should have a method for this work.